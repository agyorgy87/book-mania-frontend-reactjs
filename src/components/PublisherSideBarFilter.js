import React from 'react';

const PublisherSideBarFilter = ({options, onFilterClick}) => { 
  return (
    <div>
        <li 
            className="list-group-item name-of-the-list active border-0 rounded">Publishers</li>
            {
                options?.map((publisher, index) => (
                    <li 
                        key={"all-publisher-div" + index} 
                        className="list-group-item list-group-item-action border-0 options-for-filtering" 
                        onClick={() => onFilterClick(publisher.publisher_name)}>{publisher.publisher_name}</li>
                ))
            }
    </div>
  )
}

export default PublisherSideBarFilter;