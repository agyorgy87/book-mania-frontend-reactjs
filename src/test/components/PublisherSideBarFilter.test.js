import renderer from 'react-test-renderer';
import PublisherSideBarFilter from '../../components/PublisherSideBarFilter';

test("test with empty list", () => {
    let tree = renderer
    .create(<PublisherSideBarFilter options={[]} filter={() => {}}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});

test("test with options list", () => {
    let inspectArray = [
        {publisher_name: "dog"}
    ]
    let tree = renderer
    .create(<PublisherSideBarFilter options={inspectArray} filter={() => {}}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});

test("test with undefined", () => {
    let tree = renderer
    .create(<PublisherSideBarFilter options={undefined} filter={() => {}}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});



