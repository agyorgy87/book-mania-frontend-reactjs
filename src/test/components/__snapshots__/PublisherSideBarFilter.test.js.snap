// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with empty list 1`] = `
<div>
  <li
    className="list-group-item name-of-the-list active border-0 rounded"
  >
    Publishers
  </li>
</div>
`;

exports[`test with options list 1`] = `
<div>
  <li
    className="list-group-item name-of-the-list active border-0 rounded"
  >
    Publishers
  </li>
  <li
    className="list-group-item list-group-item-action border-0 options-for-filtering"
    onClick={[Function]}
  >
    dog
  </li>
</div>
`;

exports[`test with undefined 1`] = `
<div>
  <li
    className="list-group-item name-of-the-list active border-0 rounded"
  >
    Publishers
  </li>
</div>
`;
